[DOCS] Clarify docs: how to add Routing enforcement

cjcjameson opened the pull request:
In 6.8, the `_doc` made sense: https://www.elastic.co/guide/en/elasticsearch/reference/6.8/mapping-routing-field.html

but now it doesn't, and is confusing. I assume it applies to all documents(??)

cjcjameson commented:
I did sign before submitting; feel free to refresh the CLA verification.

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

elasticmachine commented:
Pinging @elastic/es-search (:Search/Mapping)

jrodewig commented:
HI @cjcjameson 

Can you ensure that you have signed the CLA using the same **@users.noreply.github.com** email address associated with this commit?

That's likely causing the CLA check to fail. Thanks!

jrodewig commented:
@elasticmachine test this please

jrodewig reviewed (approved):
LGTM.

I'll merge after the CLA CI test passes.

cjcjameson committed:
Clarify docs: how to add Routing enforcement

In 6.8, the `_doc` made sense: https://www.elastic.co/guide/en/elasticsearch/reference/6.8/mapping-routing-field.html

but now it doesn't, and is confusing. I assume it applies to all documents(??)

cjcjameson commented:
@jrodewig thanks! I changed the author and also repointed this MR to the `master` branch. Can you do the backporting? What's the system for backporting docs changes to the right branches?

Also filed #53991 regarding the CLA

jrodewig commented:
@cjcjameson Thanks. It looks like that solved the CLA issue.

I'm going to change the base branch back to 7.6. That'll let me merge it and backport the changes.
For this repo, the repo maintainers (me in the case) handle the redirects. I'll get this applied to doc branches we still maintain.

Thanks again!

jrodewig commented:
@elasticmachine test this please.

jrodewig commented:
Switched base branch back to master. That seemed to clear up the merge conflicts. Will merge after CI is complete.
